causal_testing/testing/causal_test_case.py

```diff
@@ -73,6 +81,45 @@ def get_treatment_value(self):
         """Return the treatment value of the treatment variable in this causal test case."""
         return self.treatment_value
 
+    def execute_test(self, estimator: type(Estimator), data_collector: ObservationalDataCollector) -> CausalTestResult:
+        """Execute a causal test case and return the causal test result.
+
+        :param estimator: A reference to an Estimator class.
+        :param causal_test_case: The CausalTestCase object to be tested
+        :return causal_test_result: A CausalTestResult for the executed causal test case.
+        """
+        if estimator.df is None:
+            estimator.df = data_collector.collect_data()
```

Comment:
Will this basically be dead code if we're calling `collect_data()` in the json frontend? If so, do we need to call it there? Could we just get away with calling it here instead?

Reply:
I agree, calling it only in `causal_test_case` makes more sense, will fix :)